proposal.html

```diff
@@ -485,7 +485,7 @@
 
   <ins class="block">
     <emu-alg>
-      1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|.
+      1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ is a <emu-xref href="#sec-well-known-intrinsic-objects">Well-known Intrinsic Object</emu-xref> and the Global Name associated with _func_ has the syntax of a |BindingIdentifier|, the portion of the returned String that would be matched by |BindingIdentifier| must be the same as the Global Name associated with _func_.
```

Comment:
> This change would be incompatible with that requirement.

Can you walk me through that? My assumption is that the change would have no observable difference, and was predicated on the assumption that built-in objects have no defined `[[SourceText]]` (which could be made explicit in step 1). In general terms,
> 1. If _func_ is not special (bound or built-in), return a SourceText-based representation.
2. If _func_ is callable, return a _NativeFunction_ representation.

replacing the current
> 1. If _func_ is bound or built-in Function, return a _NativeFunction_ representation.
2. If _func_ has \[[SourceText]], return a SourceText-based representation.
3. If _func_ is callable, return a _NativeFunction_ representation.

In other words, what differentiates steps 1 and 3?